scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/PipelineSubscriber.kt

```diff
@@ -65,9 +77,118 @@ class PipelineSubscriber(
             .keepAliveTime(60L, TimeUnit.SECONDS)
             .keepAliveTimeout(2L, TimeUnit.SECONDS)
             .build()
-    private val client = ChainerGrpcKt.ChainerCoroutineStub(channel)
 
-    private val pipelines = ConcurrentHashMap<PipelineId, Pipeline>()
+    val client = ChainerGrpcKt.ChainerCoroutineStub(channel)
+    val dispatcher = Executors.newFixedThreadPool(nThreads).asCoroutineDispatcher()
+    val scope = CoroutineScope(SupervisorJob() + dispatcher)
+
+    val pipelines = ConcurrentHashMap<PipelineId, Pipeline>()
+    private val queues = ConcurrentHashMap<PipelineId, QueueInfo>()
+    private val queuesMutex = Mutex()
+
+    // Task factory for creating pipeline operation tasks
+    private val taskFactory =
+        PipelineTaskFactory(
+            pipelineSubscriber = this,
+            kafkaAdmin = kafkaAdmin,
+            kafkaProperties = kafkaProperties,
+            kafkaDomainParams = kafkaDomainParams,
+            name = name,
+            logger = logger,
+        )
+
+    // Track queues scheduled for deletion
+    private data class QueueInfo(
+        val queue: Channel<Task>,
+        val processingJob: Job,
+        var isMarkedForDeletion: Boolean = false,
+        var deletionScheduledAt: Long = 0L,
+    )
+
+    init {
+        // Start background cleanup task
+        scope.launch {
+            while (true) {
+                delay(5000L)
```

Comment:
I think this is too aggressive (short), especially as it is a "stop the world" (control-plane wise) operation. I would do the cleanup every 5 minutes or so. I don't think the added memory consumption would be significant, even with thousands of pipelines. However, if I'm missing something let's discuss.